src/unix/linux-syscalls.c

```diff
@@ -250,7 +250,9 @@ int uv__statx(int dirfd,
 #if !defined(__NR_statx) || defined(__ANDROID_API__) && __ANDROID_API__ < 30
   return errno = ENOSYS, -1;
 #else
-  return syscall(__NR_statx, dirfd, path, flags, mask, statxbuf);
+  int ret = syscall(__NR_statx, dirfd, path, flags, mask, statxbuf);
+  uv__msan_unpoison(&statxbuf, sizeof(struct uv__statx));
```

Comment:
Hmm, maybe unpoison on success only? I'm guessing the kernel doesn't touch the buffer on failure.